src/highlighters.cc

```diff
@@ -1056,27 +1057,43 @@ struct ShowWhitespacesHighlighter : Highlighter
             return value.str();
         };
 
+        const bool has_guide_switch = parser.get_switch("guide") ? true : false;
+
         return std::make_unique<ShowWhitespacesHighlighter>(
             get_param("tab", "→"), get_param("tabpad", " "), get_param("spc", "·"),
-            get_param("lf", "¬"), get_param("nbsp", "⍽"));
+            get_param("lf", "¬"), get_param("nbsp", "⍽"), get_param("guide", " "), has_guide_switch);
     }
 
 private:
     void do_highlight(HighlightContext context, DisplayBuffer& display_buffer, BufferRange) override
     {
         const int tabstop = context.context.options()["tabstop"].get<int>();
-        auto whitespaceface = context.context.faces()["Whitespace"];
+        int indentwidth = context.context.options()["indentwidth"].get<int>();
+        if (indentwidth == 0)
+            indentwidth = tabstop;
+
+        const auto whitespaceface = context.context.faces()["Whitespace"];
+        const auto indentguideface = context.context.faces()["IndentGuide"];
+
         const auto& buffer = context.context.buffer();
-        auto win_column = context.setup.window_pos.column;
+        const auto win_column = context.setup.window_pos.column;
+        auto line_index = context.setup.window_pos.line;
+        const LineCount buffer_end = buffer.end().coord().line;
+
         for (auto& line : display_buffer.lines())
         {
+            bool only_whitespace = m_add_guide and
+                                   line_index < buffer_end ? all_whitespace(buffer[line_index++], win_column) : true;
```

Comment:
was getting segfaults on buffer[line_index++] until I added this check. Feel like their should be a cleaner way to do this though.

Reply:
Actually, this is probably because of line wrapping isn't it. I think current implementation is looking at the wrong buffer lines for all lines after a wrapped one.